File: src/main/scala/JavaProject/ModelChoice.scala
package JavaProject

import java.io.File
import epic.sequences.{SemiCRF, TaggedSequence}

object ModelChoice {

    /* Set up all calculations, like value of least conf,
    * information density, risk so on.
    * Set up a system to call and combine methods at will.
     */

    def getValueModel(model : SemiCRF[String,String], choice: String, sentence: String):Double = {
      val words = sentence.split(" ").toSeq
        if (choice.toLowerCase().equals("lc")) {//Least Confidence
            val conf = model.leastConfidence(words.to)
            return -conf
        }
        else if (choice.toLowerCase().equals("gibbs") ){
            val posteriors = model.getPosteriors(words.to)
          var sum = 0.0
            for (i <- 0 until posteriors.length)
              {
                sum += posteriors(i)* posteriors(i)
              }
          if(sum>1||sum<0) {
            println("Gibbs sum is: " + sum)
          }
          return 1- sum
        }
        else {
            return 0;
        }
    }
}

File: src/main/scala/JavaProject/ModelChoice.java
import java.io.*;
import java.util.*;
import java.lang.Object;

public class ModelChoice
{
    private final int choice;

    public ModelChoice(int choice, String fileName){
        this.choice = choice;
    }

    /* Set up all calculations, like value of least conf,
    * information density, risk so on.
    * Set up a system to call and combine methods at will.
     */

    public double getValueModel(Object list) {

        if( choice == 1)
        {
            String line = list.toString();
            return line.length();
        }
        else {
            return 0;
        }
    }

}